corehq/apps/users/models.py

```diff
@@ -1139,6 +1139,33 @@ def get(stale, raise_if_none):
         else:
             return None
 
+    @classmethod
+    @skippable_quickcache(['username'], skip_arg='strict')
+    def get_deleted_by_username(cls, username, strict=True):
+        def get(stale, raise_if_none):
+            result = cls.get_db().view('deleted_users/by_username',
+                                       key=username,
+                                       include_docs=True,
+                                       reduce=False,
+                                       stale=stale if not strict else None,
+                                       )
+            return result.one(except_all=raise_if_none)
+
+        try:
+            result = get(stale=settings.COUCH_STALE_QUERY, raise_if_none=True)
+            if result['doc'] is None or result['doc']['username'] != username:
+                raise NoResultFound
+        except NoMoreData:
+            logging.exception('called get_by_username(%r) and it failed pretty bad' % username)
```

Comment:
i don't think this log message is particularly useful compared to the stack trace and since we don't try and handle this scenario it's essentially the same as just letting it fail hard (which i think is fine in this case). basically saying you don't know this except clause